### ISSUE TITLE
Git commit id is hard to find

### ISSUE THREAD

**Issue by mborgerson** [xqemu/xqemu#47]
Suggest adding it to title bar of XQEMU window

**Comment by MayeulC** [comment]
You can do something like the following in the compilation script:

```bash
COMMIT_ID=$(git log --format=%H -n 1|| echo unknown rev)

# and in :
--extra-cflags=" [...] -DGIT_COMMIT_ID=G_STRINGIFY\\($COMMIT_ID\\) \
```

This way you can use the following in the code:
```C
fprintf("git revision is %s\n", GIT_COMMIT_ID)
```

Note that this depends on `G_STRINGIFY` [from glibc](https://developer.gnome.org/glib/stable/glib-Miscellaneous-Macros.html#G-STRINGIFY:CAPS). If necessary, you can redefine it as:

```C
#define G_STRINGIFY(macro_or_string) G_STRINGIFY_ARGS(macro_or_string)
#define G_STRINGIFY_ARGS(string) #string
```
Two steps are necessary for the `macro_or_string` macro expansion.

I don't like much the fact that I had to escape the parenthesis, though. I tried to work around it, but could not find a way.
Note that if `make` is called directly, the commit id won't be updated. The above code prints "unknown rev" if git is not installed on the system.

**Comment by LukeUsher** [comment]
We handle this in Cxbx-Reloaded by having a build script generate a Version.h header file with the git version string as a define. That way it has no dependencies other than Git being installed on the machine doing the compilation.

**Comment by mborgerson** [comment]
Thanks guys! Qemu actually already does this for us as part of its build process so we can probably use that. It's basically what you've both suggested. The `make` process will generate a file called qemu-version.h that looks like:

    #define QEMU_PKGVERSION "v2.12.0-31-g7378eee5f2-dirty"
    #define QEMU_FULL_VERSION QEMU_VERSION " (" QEMU_PKGVERSION ")"

**Comment by dracc** [comment]
Closed by #57

**Comment by mborgerson** [comment]
Thanks @dracc
